watertap/property_models/tests/test_multicomp_aq_sol_prop_pack.py

```diff
@@ -2485,3 +2507,625 @@ def test_automatic_charge_mw_population():
     for comp, val in test_vals.items():
         assert value(m.fs.stream2[0].mw_comp[comp]) == val[0]
         assert value(m.fs.stream2[0].charge_comp[comp]) == val[1]
+
+
+class TestMCASInitializer:
+    @pytest.mark.component
+    def test_initialize(self):
+        m = ConcreteModel()
+        m.fs = FlowsheetBlock(dynamic=False)
+        m.fs.properties = MCASParameterBlock(
+            solute_list=["Ca_2+", "SO4_2-", "Na_+", "Cl_-", "Mg_2+"],
+            material_flow_basis=MaterialFlowBasis.mass,
+            diffusivity_data={
+                ("Liq", "Ca_2+"): 7.92e-10,
+                ("Liq", "SO4_2-"): 1.06e-09,
+                ("Liq", "Na_+"): 1.33e-09,
+                ("Liq", "Cl_-"): 2.03e-09,
+                ("Liq", "Mg_2+"): 7.06e-10,
+            },
+            mw_data={
+                "H2O": 0.018,
+                "Na_+": 0.023,
+                "Ca_2+": 0.04,
+                "Mg_2+": 0.024,
+                "Cl_-": 0.035,
+                "SO4_2-": 0.096,
+            },
+            stokes_radius_data={
+                "Na_+": 1.84e-10,
+                "Ca_2+": 3.09e-10,
+                "Mg_2+": 3.47e-10,
+                "Cl_-": 1.21e-10,
+                "SO4_2-": 2.3e-10,
+            },
+            charge={"Na_+": 1, "Ca_2+": 2, "Mg_2+": 2, "Cl_-": -1, "SO4_2-": -2},
+            elec_mobility_calculation=ElectricalMobilityCalculation.EinsteinRelation,
+            density_calculation=DensityCalculation.seawater,
+            activity_coefficient_model=ActivityCoefficientModel.davies,
+        )
+
+        m.fs.stream = stream = m.fs.properties.build_state_block(
+            [0], defined_state=True
+        )
+
+        mass_flow_in = 1 * pyunits.kg / pyunits.s
+        feed_mass_frac = {
+            "Na_+": 11122e-6,
+            "Ca_2+": 382e-6,
+            "Mg_2+": 1394e-6,
+            "SO4_2-": 2136e-6,
+            "Cl_-": 20300e-6,
+        }
+        for ion, x in feed_mass_frac.items():
+            mass_comp_flow = x * pyunits.kg / pyunits.kg * mass_flow_in
+
+            stream[0].flow_mass_phase_comp["Liq", ion].fix(mass_comp_flow)
+
+        H2O_mass_frac = 1 - sum(x for x in feed_mass_frac.values())
+
+        stream[0].flow_mass_phase_comp["Liq", "H2O"].fix(H2O_mass_frac)
+        stream[0].temperature.fix(298.15)
+        stream[0].pressure.fix(101325)
+
+        stream[0].assert_electroneutrality(
+            defined_state=True, tol=1e-2, adjust_by_ion="Cl_-"
+        )
+
+        metadata = m.fs.properties.get_metadata().properties
+        for v in metadata.list_supported_properties():
+            getattr(stream[0], v.name)
+            assert stream[0].is_property_constructed(v.name)
+
+        initializer = stream.default_initializer()
+        initializer.initialize(stream)
```

Comment:
I am not quite sure what you would be asserting in this case. What is the outcome of an initialization in the general sense? One might say that it is converging the model to a given tolerance, but that is not quite true as it only needs to get it to the point that a full solver can take over (although many of our routines include that as the final step).

Reply:
TBH- this thought was geared more towards the Scaler originally, but then I figured that we should think about doing some form of a check on Initializers although it is not apparent what the best way to go about it would be, or if it would be worthwhile for something like a property block.  

One idea- even though it may not be the best approach- is to check the approximate number of iterations before convergence, which could be a pain between the different runners. Something like that, or we can set a low number on max_iter and check the state of the model. But again--now that I think about it- this probably makes sense for a flowsheet or a complex model and not so much for a more basic model.

Overall though, the main question that I wanted to answer was -- how can we compare two initializers/initialization methods at this level to decide on whether one is superior (for the tested use case)?

Reply:
That is not easy to define. For one, determining the number of iterations is difficult to start with (as the solvers do not report it), and sometime more iterations are acceptable if it results in greater robustness across a range of conditions. That perhaps is the bigger point - it is hard to mark one initializer as superior to another without considering what they were designed to do (and this is ultimately why IDAES moved to class-based initializers to that we could have multiple options if required).

One thing I can say however is that the number of iterations should be fairly consistent across runners if you have a well scaled problem (but bets are off is the scaling is not good).